lib/cisco_node_utils/syslog_server.rb

```diff
@@ -42,12 +52,20 @@ def initialize(name,
 
     def self.syslogservers
       hash = {}
-
       syslogservers_list = config_get('syslog_server', 'server')
       return hash if syslogservers_list.nil?
 
       syslogservers_list.each do |id|
+        # The YAML regex isn't specific enough for some platforms,
+        # so we have to do further checking.
+        begin
+          IPAddr.new(id)
+        rescue
+          next
+        end
+
         level = config_get('syslog_server', 'level', id)
+        level = LEVEL_TO_NUM[level] if platform == :ios_xr
 
         vrf = config_get('syslog_server', 'vrf', id)
```

Comment:
Ah, I see. My apologies for not looking closer at the netdev types. So, do I understand correctly that netdev_stdlib expects that:

1) the id (IP or name) is a unique identifier
2) the VRF is a property of a server but does not distinguish between servers

If so, then I think the correct thing to do in this code to yield the expected behavior would be to:
a) have it log a warning (or raise an error) if in iterating over the servers it finds the same server in multiple VRFs
b) in create(), remove any existing server entry in a different VRF before creating the entry in the given VRF.

Does this sound more reasonable?